src/front/wgsl/mod.rs

```diff
@@ -4723,28 +3796,2227 @@ impl Parser {
         }
     }
 
-    pub fn parse(&mut self, source: &str) -> Result<crate::Module, ParseError> {
+    pub fn parse<'a>(&mut self, source: &'a str) -> Result<TranslationUnit<'a>, ParseError> {
```

Comment:
Let's keep the fn signature the same considering this is part of our public API and there isn't much people can do with a `TranslationUnit<'a>`.

This means we don't have to expose `Lowerer` and we can undo the changes to `parse_str` & the benchmark file.